openupgrade_scripts/scripts/account/15.0.1.2/pre-migration.py

```diff
@@ -0,0 +1,409 @@
+                            aml.credit != 0 AND aml.credit IS NOT NULL))
+                        OR (tax.type_tax_use = 'sale' AND (
+                            aml.debit = 0 OR aml.debit IS NULL))
+                    THEN TRUE
+                    ELSE FALSE
+                END AS tax_tag_invert
+            FROM account_move am
+            JOIN account_move_line aml ON am.id = aml.move_id
+            JOIN account_move_line_account_tax_rel tax_rel ON
+                aml.id = tax_rel.account_move_line_id
+            JOIN account_tax tax ON tax.id = tax_rel.account_tax_id
+            WHERE am.move_type = 'entry'
+        ) sub
+        WHERE sub.id = aml.id AND aml.tax_tag_invert IS NULL""",
+    )
+
+
+def _create_account_payment_method_line(env):
+    # Create account_payment_method_line table
+    sql.create_model_table(
+        env.cr,
+        "account_payment_method_line",
+        columns=[
+            ("name", "varchar", ""),
+            ("sequence", "integer", ""),
+            ("payment_method_id", "integer", ""),
+            ("payment_account_id", "integer", ""),
+            ("journal_id", "integer", ""),
+            ("create_uid", "integer", ""),
+            ("create_date", "timestamp", ""),
+            ("write_uid", "integer", ""),
+            ("write_date", "timestamp", ""),
+        ],
+    )
+    # Create account payment method lines from account payment methods
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO account_payment_method_line (name, sequence,
+            payment_method_id, journal_id, create_uid, write_uid,
+            create_date, write_date)
+        SELECT apm.name, 10, apm.id, aj.id,
+            apm.create_uid, apm.write_uid, apm.create_date, apm.write_date
+        FROM account_payment_method apm, account_journal aj
+        WHERE apm.code = 'manual' AND aj.type IN ('bank', 'cash')
+        """,
+    )
+
+
+def _fast_fill_account_payment_payment_method_line_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE account_payment
+        ADD COLUMN IF NOT EXISTS payment_method_line_id INTEGER""",
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_payment ap
+        SET payment_method_line_id = apml.id
+        FROM account_move am
+        JOIN account_payment_method_line apml ON apml.journal_id = am.journal_id
+        WHERE ap.move_id = am.id AND ap.payment_method_id = apml.payment_method_id
+        """,
+    )
+
+
+def _fill_account_tax_country_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE account_tax
+        ADD COLUMN IF NOT EXISTS country_id INTEGER
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_tax at
+        SET country_id = p.country_id
+        FROM res_company c
+        JOIN res_partner p ON p.id = c.partner_id
+        WHERE c.id = at.company_id
```

Comment:
I would rename `res_company.account_tax_fiscal_country_id` to the new `account_fiscal_country_id` column and use that here instead of `res_company.country_id`